hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -1360,4 +1360,22 @@ private Constants() {
    * Value: {@value}.
    */
   public static final boolean OPTIMIZED_COPY_FROM_LOCAL_DEFAULT = true;
+
+
+  /**
+   * To use the same classloader that loaded S3AFileSystem to al
+   * load the user extensions, such as {{fs.s3a.aws.credentials.provider}}.
+   * It is useful to turn this off for Apache Spark applications that
+   * might load S3AFileSystem from the Spark distribution (Launcher classloader)
+   * while users might want to provide custom extensions (loaded by Spark MutableClassloader).
+   * Default value: true.
+   */
+  public static final String AWS_S3_EXTENSIONS_ISOLATED_CLASSLOADER =
```

Comment:
I don't know if this is the right name, it's not really an extension (we use fs.s3a.ext for "any plugin can make up a name and we won't do any in the fs)

maybe `fs.s3a.classloader.isolation`